test-openjdk/src/test/java/com/redhat/qe/openjdk/jlink/TemplateBuildTest.java

```diff
@@ -0,0 +1,229 @@
+        DEFAULT_TEMPLATE_PARAMS.put("IMAGESTREAM_NAMESPACE", NamespaceManager.getNamespace());
+        DEFAULT_TEMPLATE_PARAMS.put("CONTEXT_DIR", "quarkus-quickstarts/getting-started-3.9.2-uberjar");
+        DEFAULT_TEMPLATE_PARAMS.put("APP_URI", "https://github.com/rh-openjdk/openjdk-container-test-applications");
+        DEFAULT_TEMPLATE_PARAMS.put("REF", "master");
+        DEFAULT_TEMPLATE_PARAMS.put("SERVICE_PORT", "8080");
+        DEFAULT_TEMPLATE_PARAMS.put("TARGET_PORT", "8080");
+        DEFAULT_TEMPLATE_PARAMS.put("BUILDER_IMAGE_TAG", IMAGE_BUILDER_TAG);
+    }
+
+    @BeforeEach
+    public void prepareForTests() throws IOException, InterruptedException {
+        oc.loginDefault();
+        oc.project(NamespaceManager.getNamespace());
+
+        //prerequisites - provided images must be different for the tests to work
+        Assertions.assertThat(IMAGE_BUILDER_URL).isNotEmpty();
+
+        createImageStream(IMAGE_BUILDER_URL, IMAGE_BUILDER_IMAGESTREAM, IMAGE_BUILDER_TAG);
+
+        ensureBuildTemplateIsInstalled();
+    }
+
+    @Test
+    public void validateRequiredParams() {
+        SoftAssertions soft = new SoftAssertions();
+
+        Map<String, Boolean> expectedParamsRequiredMap = new HashMap<>();
+        expectedParamsRequiredMap.put("JDK_VERSION", true);
+        expectedParamsRequiredMap.put("BUILDER_IMAGE_TAG", false);
+        expectedParamsRequiredMap.put("APPNAME", true);
+        expectedParamsRequiredMap.put("APP_URI", true);
+        expectedParamsRequiredMap.put("REF", true);
+        expectedParamsRequiredMap.put("CONTEXT_DIR", true);
+        expectedParamsRequiredMap.put("GITHUB_WEBHOOK_SECRET", true);
+        expectedParamsRequiredMap.put("TARGET_PORT", true);
+        expectedParamsRequiredMap.put("SERVICE_PORT", true);
+
+        Template buildTemplate = openShift.getTemplate(BUILD_TEMPLATE_NAME);
+        List<Parameter> templateParameters = buildTemplate.getParameters();
+        List<String> paramNames = templateParameters.stream().map(Parameter::getName).collect(Collectors.toList());
+
+        soft.assertThat(paramNames).as("The actual parameters differ from the expected set")
+                .containsExactlyInAnyOrderElementsOf(expectedParamsRequiredMap.keySet());
+        soft.assertThat(templateParameters).as("The actual set of required parameters differs from the expected set")
+                .allMatch(p -> (p.getRequired() == null && !expectedParamsRequiredMap.get(p.getName()))
+                        || expectedParamsRequiredMap.get(p.getName()) == p.getRequired());
+
+        soft.assertAll();
+    }
+
+    @Test
+    public void validateBuildFromTemplate() {
+        Map<String, String> params = new HashMap<>(DEFAULT_TEMPLATE_PARAMS);
+        useTemplateToCreateResources(params);
+
+        verifyExpectedResourcesCreated();
+        // Validate the app is up
+        assertTrue(Https.doesUrlReturnCode("http://" + HOSTNAME, 200).waitFor());
+        System.out.println("Statement.");
```

Comment:
Statement? Just statement?-)

Reply:
I need to remove that. I was using it so I could study the assert with the debugger